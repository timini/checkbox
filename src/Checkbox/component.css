.flex {
  display: flex;
  align-items: center;
}

.black {
  color: black;
}

.yellow {
  color: #FFD700;
}

.pink {
  color: #D5008F;
}

.green {
  color: #137752;
}

.blue {
  color: #00449E;
}
