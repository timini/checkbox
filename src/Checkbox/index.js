import Checkox from "./component";

export default Checkox;
