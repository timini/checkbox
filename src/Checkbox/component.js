import React, { PureComponent } from "react";
import PropTypes from "prop-types";
import { equals, type } from "ramda";
import "./component.css";

const SIZES = {
  small: 0.3,
  medium: 0.5,
  large: 1,
  "x-large": 2
};

const length = size => n => `${SIZES[size] * n}`;

export class Checkbox extends PureComponent {
  static propTypes = {
    id: PropTypes.string.isRequired,
    checked: PropTypes.bool.isRequired,
    disabled: PropTypes.bool,
    label: PropTypes.string,
    colour: PropTypes.string,
    onClick: PropTypes.func.isRequired
  };
  static defaultProps = {
    checked: false,
    disabled: false,
    label: "",
    size: "medium",
    colour: "black"
  };
  constructor(props) {
    super(props);
    this.onClick = this.onClick.bind(this);
  }
  onClick(event) {
    const { id, onClick, disabled, checked: prevChecked } = this.props;
    if (disabled) return;
    const checked = !prevChecked;
    if (equals(type(onClick), "Function")) {
      onClick(id, checked);
    }
  }
  render() {
    const { label, disabled, checked, size, colour } = this.props;
    const len = length(size);
    return (
      <div onClick={this.onClick} className={colour}>
        <svg
          width={len(48)}
          height={len(48)}
          viewBox={`0 0 ${len(48)} ${len(48)}`}
          style={{ marginBottom: len(-6) }}
        >
          <rect
            width={len(48)}
            height={len(48)}
            className="border"
            stroke={colour}
            fill="white"
            strokeWidth={len(8)}
          />
          {checked ? (
            <path
              d={`M${len(16)},${len(32)}l${len(3)},${len(3)}L${len(37)},${len(
                17
              )}l-${len(3)}-${len(3)}Z`}
            />
          ) : null}
          {checked ? (
            <path
              d={`M${len(22)},${len(32)}l-${len(3)},${len(3)}-${len(8)}-${len(
                8
              )},${len(3)}-${len(3)}Z`}
            />
          ) : null}
        </svg>
        {label && (
          <span style={{ marginLeft: `0.25em`, fontSize: `${len(3)}em` }}>
            {label}
          </span>
        )}
      </div>
    );
  }
}

export default Checkbox;
