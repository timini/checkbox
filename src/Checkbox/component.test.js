import React from "react";
import { shallow } from "enzyme";
import Checkbox from "./component";

it("should render correctly", () => {
  const wrapper = shallow(<Checkbox label="test"/>);
  expect(wrapper.find("span[children='test']").text()).toEqual('test');
  expect(wrapper).toMatchSnapshot();
});

it("should call the onClick handler", () => {
  const onClick = jest.fn();
  const wrapper = shallow(<Checkbox id="1" label="test" onClick={onClick}/>);
  wrapper.find('div').simulate('click')
  expect(onClick).toHaveBeenCalledWith("1", true);
})

it("should not call the onClick handler when disabled", () => {
  const onClick = jest.fn();
  const wrapper = shallow(<Checkbox id="1" label="test" onClick={onClick} disabled/>);
  wrapper.find('div').simulate('click')
  expect(onClick).not.toHaveBeenCalled();
})
