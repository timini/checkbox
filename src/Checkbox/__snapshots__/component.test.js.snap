// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Checkbox
    checked={false}
    colour="black"
    disabled={false}
    label="test"
    size="medium"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <svg
          height="24"
          style={
                    Object {
                              "marginBottom": "-3",
                            }
          }
          viewBox="0 0 24 24"
          width="24"
>
          <rect
                    className="border"
                    fill="white"
                    height="24"
                    stroke="black"
                    strokeWidth="4"
                    width="24"
          />
</svg>,
        <span
          style={
                    Object {
                              "fontSize": "1.5em",
                              "marginLeft": "0.25em",
                            }
          }
>
          test
</span>,
      ],
      "className": "black",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <rect
              className="border"
              fill="white"
              height="24"
              stroke="black"
              strokeWidth="4"
              width="24"
/>,
            null,
            null,
          ],
          "height": "24",
          "style": Object {
            "marginBottom": "-3",
          },
          "viewBox": "0 0 24 24",
          "width": "24",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "border",
              "fill": "white",
              "height": "24",
              "stroke": "black",
              "strokeWidth": "4",
              "width": "24",
            },
            "ref": null,
            "rendered": null,
            "type": "rect",
          },
          null,
          null,
        ],
        "type": "svg",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "test",
          "style": Object {
            "fontSize": "1.5em",
            "marginLeft": "0.25em",
          },
        },
        "ref": null,
        "rendered": "test",
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <svg
            height="24"
            style={
                        Object {
                                    "marginBottom": "-3",
                                  }
            }
            viewBox="0 0 24 24"
            width="24"
>
            <rect
                        className="border"
                        fill="white"
                        height="24"
                        stroke="black"
                        strokeWidth="4"
                        width="24"
            />
</svg>,
          <span
            style={
                        Object {
                                    "fontSize": "1.5em",
                                    "marginLeft": "0.25em",
                                  }
            }
>
            test
</span>,
        ],
        "className": "black",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <rect
                className="border"
                fill="white"
                height="24"
                stroke="black"
                strokeWidth="4"
                width="24"
/>,
              null,
              null,
            ],
            "height": "24",
            "style": Object {
              "marginBottom": "-3",
            },
            "viewBox": "0 0 24 24",
            "width": "24",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "border",
                "fill": "white",
                "height": "24",
                "stroke": "black",
                "strokeWidth": "4",
                "width": "24",
              },
              "ref": null,
              "rendered": null,
              "type": "rect",
            },
            null,
            null,
          ],
          "type": "svg",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "test",
            "style": Object {
              "fontSize": "1.5em",
              "marginLeft": "0.25em",
            },
          },
          "ref": null,
          "rendered": "test",
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
